### ISSUE TITLE
[Consideration] STX as Result/Error library

### ISSUE THREAD

**Issue by redradist** [bytecodealliance/wasmtime-cpp#2]
For me it seems like it would be useful to use [STX](https://github.com/lamarrr/STX) library as Result/Error library
This library inspired by Rust `Result`
Also this library available in `conan-center`

**Comment by alexcrichton** [comment]
In theory this API depends only a little bit on `Result`, so I'd ideally prefer to avoid taking a whole dependency just for this. Perhaps though the `wasmtime::Result` type could be copied over from the stx version? (or at least the most important bits)

**Comment by redradist** [comment]
Yeah, I agree maybe it is not necessary to bring bit dependency ...
